src/test.cc

```diff
@@ -75,10 +75,11 @@ char* mkdtemp(char* name_template) {
 
 string GetSystemTempDir() {
 #ifdef _WIN32
-  char buf[1024];
-  if (!GetTempPath(sizeof(buf), buf))
+  WCHAR buf[MAX_PATH];
+  // GetTempPath returns zero if no path is found
+  if(GetTempPathW(MAX_PATH, buf) == 0)
     return "";
-  return buf;
+  return WideToUtf8(wstring(buf));
```

Comment:
Two allocations, and two copies.

Modify WideToUtf8 so that the input is a StringPiece, allowing you to avoid one allocation and copy.

Reply:
How about removing the inner allocation of the wstring? Shouldn't the return just be a move command?

Reply:
> How about removing the inner allocation of the wstring?

Yes, otherwise using a WStringPiece as input wouldn't make sense.

> Shouldn't the return just be a move command?

Yes, or RVO should kick in.

Reply:
WideToUtf8 currently takes a `std::wstring const&`. If you pass `buf` to it, you will get an allocation and copy.

You *must* fix your conversion functions to resolve most of my complaints about unnecessary allocations and copies.